src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/Dac/Utils/DacPropertySymbolUtils.cs

```diff
@@ -0,0 +1,273 @@
+﻿using Acuminator.Utilities.Common;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+
+namespace Acuminator.Utilities.Roslyn.Semantic.Dac
+{
+	public static class DacPropertySymbolUtils
+	{
+		/// <summary>
+		/// A delegate type for a DAC property which extracts info DTOs about DAC properties/fields from <paramref name="dacOrDacExtension"/> 
+		/// and adds them to the <paramref name="dacInfos"/> collection with a consideration for DAC properties/fields declaration order.
+		/// Returns the number following the last assigned declaration order.
+		/// </summary>
+		/// <typeparam name="TInfo">Generic type parameter representing overridable info type.</typeparam>
+		/// <param name="dacPropertyInfos">The DAC property infos.</param>
+		/// <param name="dacOrDacExtension">The DAC or DAC extension.</param>
+		/// <param name="startingOrder">The declaration order which should be assigned to the first DTO.</param>
+		/// <returns/>
+		private delegate int AddDacPropertyInfoWithOrderDelegate<TInfo>(OverridableItemsCollection<TInfo> dacPropertyInfos,
+																		ITypeSymbol dacOrDacExtension, int startingOrder)
+		where TInfo : IOverridableItem<TInfo>;
+
+
+		#region Dac Properties
+		/// <summary>
+		/// Gets the DAC property symbols and syntax nodes from DAC and, if <paramref name="includeFromInheritanceChain"/> is <c>true</c>, its base DACs.
+		/// </summary>
+		/// <param name="dac">The DAC to act on.</param>
+		/// <param name="pxContext">Context.</param>
+		/// <param name="includeFromInheritanceChain">(Optional) True to include, false to exclude the properties from the inheritance chain.</param>
+		/// <param name="cancellation">(Optional) Cancellation token.</param>
+		/// <returns>The DAC property symbols with nodes from DAC.</returns>
+		public static OverridableItemsCollection<DacPropertyInfo> GetDacPropertiesFromDac(this ITypeSymbol dac, PXContext pxContext,
+																						  bool includeFromInheritanceChain = true,
+																						  CancellationToken cancellation = default)
+		{
+			pxContext.ThrowOnNull(nameof(pxContext));
+
+			if (!dac.IsDAC(pxContext))
+				return new OverridableItemsCollection<DacPropertyInfo>();
```

Comment:
Why not to throw error here?

Reply:
Well I think there are two approaches with utility methods when there is an invalid input:
1) throw an exception
2) do nothing and quickly return.

There are examples of both in Acuminator and in frankly for this class I copied the logic from the similar helper for PXGraph Actions. So, one part of the answer is - to be consistent with the helpers for graph semantic model.

But also I'm not sure that first approach is always the best if we are not passing something completely invalid like null reference. 
This is an utility method so it is expected to be called in different contexts. Maybe in one place there should be an exception thrown. But the caller can throw it by himself if needed. 
On the other hand, if someone needs a silent behaviour then he is left with no choice but use try/catch.
Maybe the best approach is to use something like TryXXX pattern which will also return a flag of success to allow the caller to distinguish between failure and successfull empty result. But it is not very convenient or elegant to use such helper. 

Overall there are some pros and cons for different approaches. I suggest to stay with a "silent" one to be consistent with graph semantic utils and most of existing utils. 
If we really want to throw error then it should be consistent, so other helpers should be rewritten too.